integrated_channels/degreed/models.py

```diff
@@ -0,0 +1,273 @@
+from simple_history.models import HistoricalRecords
+
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+
+LOGGER = getLogger(__name__)
+
+
+@python_2_unicode_compatible
+class DegreedGlobalConfiguration(ConfigurationModel):
+    """
+    The global configuration for integrating with Degreed.
+    """
+
+    degreed_base_url = models.CharField(
+        max_length=255,
+        verbose_name="Degreed Base URL",
+        help_text="The base URL used for API requests to Degreed, i.e. https://degreed.com."
+    )
+
+    completion_status_api_path = models.CharField(
+        max_length=255,
+        verbose_name="Completion Status API Path",
+        help_text="The API path for making completion POST/DELETE requests to Degreed."
+    )
+
+    course_api_path = models.CharField(
+        max_length=255,
+        verbose_name="Course Metadata API Path",
+        help_text="The API path for making course metadata POST/DELETE requests to Degreed."
+    )
+
+    oauth_api_path = models.CharField(
+        max_length=255,
+        verbose_name="OAuth API Path",
+        help_text=(
+            "The API path for making OAuth-related POST requests to Degreed. "
+            "This will be used to gain the OAuth access token which is required for other API calls."
+        )
+    )
+
+    degreed_user_id = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="Degreed User ID",
+        help_text=(
+            "The Degreed User ID provided to the content provider by Degreed. "
+            "It is required for getting the OAuth access token."
+        )
+    )
+
+    degreed_user_password = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="Degreed User Password",
+        help_text=(
+            "The Degreed User Password provided to the content provider by Degreed. "
+            "It is required for getting the OAuth access token."
+        )
+    )
+
+    provider_id = models.CharField(
+        max_length=100,
+        default='EDX',
+        verbose_name="Provider Code",
+        help_text="The provider code that Degreed gives to the content provider."
+    )
+
+    class Meta:
+        app_label = 'degreed'
+
+    def __str__(self):
+        """
+        Return a human-readable string representation of the object.
+        """
+        return "<DegreedGlobalConfiguration with id {id}>".format(id=self.id)
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class DegreedEnterpriseCustomerConfiguration(EnterpriseCustomerPluginConfiguration):
+    """
+    The Enterprise specific configuration we need for integrating with Degreed.
+    """
+
+    key = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="API Client ID",
```

Comment:
Yeah, there is a mismatch, but it's mostly carry-over from the SAPSF integrated channel terminology used for the same thing. It took me some careful looking to see what `key` really meant, too. I can change it, but it might make the code harder to follow now between the 2 channels, and one of my goals has been to keep them quite consistent looking.

If you have some other idea or efficient way of doing this, do let me know.

Reply:
> I can change it, but it might make the code harder to follow now between the 2 channels, and one of my goals has been to keep them quite consistent looking.

I'm 👍 on that. No need to make this particular PR bigger than it already is!